src/manager.cpp

```diff
@@ -210,12 +215,42 @@ types::DbusVariantType
     Manager::readKeyword(const types::Path i_fruPath,
                          const types::ReadVpdParams i_paramsToReadData)
 {
-    // Code to supress unused variable warning. To be removed.
-    (void)i_fruPath;
-    (void)i_paramsToReadData;
+    try
+    {
+        nlohmann::json l_jsonObj{};
 
-    // On success return the value read. On failure throw error.
-    return types::DbusVariantType();
+        if (m_worker.get() != nullptr)
+        {
+            l_jsonObj = m_worker->getSysCfgJsonObj();
+        }
+
+        std::error_code ec;
+
+        // Check if given path is filesystem path
+        if (!std::filesystem::exists(i_fruPath, ec) && (ec))
+        {
+            throw std::runtime_error("Given file path " + i_fruPath +
+                                     " not found.");
+        }
+
+        logging::logMessage("Performing VPD read on " + i_fruPath);
+
+        std::shared_ptr<vpd::Parser> l_parserObj =
```

Comment:
Suggestion in case you want to. 

 ```
std::shared_ptr<vpd::Parser> l_parserObj =
            std::make_shared<vpd::Parser>(i_fruPath, l_jsonObj);
        
        std::shared_ptr<vpd::ParserInterface> l_vpdParserInstance =
            l_parserObj->getVpdParserInstance();

        return (
            l_vpdParserInstance->readKeywordFromHardware(i_paramsToReadData)); 
```

can be written as 
`
return((std::make_shared<vpd::Parser>(i_fruPath, l_jsonObj))->getVpdParserInstance())->readKeywordFromHardware(i_paramsToReadData))`

because you are using all those objects only once that too just call an API.
Your call.

Reply:
if we have in separate lines, the code will be more readable and easier to understand.